website/blog/2024-03-11-AutoDefense/Defending LLMs Against Jailbreak Attacks with AutoDefense.mdx

```diff
@@ -0,0 +1,110 @@
+4. **Mixtral**: Mixtral-8x7b-v0.1, Mistral-7b-v0.2
+
+We use llama-cpp-python to serve the chat completion API for open-source LLMs, allowing each LLM agent to perform inference through a unified API. INT8 quantization is used for efficiency.
+
+LLM temperature is set to `0.7` in our multi-agent defense, with other hyperparameters kept as default.
+
+## Experiment Results
+
+We design experiments to compare AutoDefense with other defense methods and different numbers of agents.
+
+![table-compared-methods](imgs/table-compared-methods.png)
+
+We compare different methods for defending GPT-3.5 as shown in Table 3. The LLaMA-2-13B is used as the defense LLM in AutoDefense. We find our AutoDefense outperforms other methods in terms of Attack Success Rate (ASR; lower is better).
+
+### Number of Agents vs Attack Success Rate (ASR)
+
+![table-agents](imgs/table-agents.png)
+
+Increasing the number of agents generally improves defense performance, especially for LLaMA-2 models. The three-agent defense system achieves the best balance of low ASR and False Positive Rate. For LLaMA-2-13b, the ASR reduces from 9.44% with a single agent to 7.95% with three agents.
+
+### Comparisons with Other Defenses
+
+AutoDefense outperforms other methods in defending GPT-3.5. Our three-agent defense system with LLaMA-2-13B reduces the ASR on GPT-3.5 from 55.74% to 7.95%, surpassing the performance of System-Mode Self-Reminder (22.31%), Self Defense (43.64%), OpenAI Moderation API (53.79%), and Llama Guard (21.28%).
+
+## Custom Agent: Llama Guard
+
+While the three-agent defense system with LLaMA-2-13B achieves a low ASR, its False Positive Rate on LLaMA-2-7b is relatively high. To address this, we introduce Llama Guard as a custom agent in a 4-agents system.
+
+Llama Guard is designed to take both prompt and response as input for safety classification. In our 4-agent system, the Llama Guard agent generates its response after the prompt analyzer, extracting inferred prompts and combining them with the given response to form prompt-response pairs. These pairs are then passed to Llama Guard for safety inference.
+
+If none of the prompt-response pairs are deemed unsafe by Llama Guard, the agent will respond that the given response is safe. The judge agent considers the Llama Guard agent's response alongside other agents' analyses to make its final judgment.
+
+As shown in Table 4, introducing Llama Guard as a custom agent significantly reduces the False Positive Rate from 37.32% to 6.80% for the LLaMA-2-7b based defense, while keeping the ASR at a competitive level of 11.08%. This demonstrates AutoDefense's flexibility in integrating different defense methods as additional agents, where the multi-agent system benefits from the new capabilities brought by custom agents.
+
+![table-4agents](imgs/table-4agents.png)
+
+## For Further Reading
+We have included samples of our prompt designs in the paper appendix below.
```

Comment:
Please refer to the further reading section in https://microsoft.github.io/autogen/blog/2023/11/09/EcoAssistant

I think we can follow that.